### ISSUE TITLE
Godot Crash When Creating new project

### ISSUE THREAD

**Issue by hackerManik3** [godotengine/godot#56076]
### Godot version

v4.0.dev.20211210.official [21b81998b]

### System information

Device Inspiron 15-3565 ,Windows 10, CPU: AMD A6-9225 2.6Gh ,GPU:RADEON R4(Built in),Ram: 4 GB

### Issue description

When I create new project it crashes...
When I open an existing 2d project it loads and work fine until I Chick 3d viewport or add a 3d object in scene which leads to opening of 3d viewport...
Work around i found include opening a 2d project and then adding a WorldEnvironmentNode to 2d scene and changing it to Clear color or Custom Color...

errors  :

this comes more often and crashes frees my cursor for 5sec : 

                ERROR: vmaMapMemory failed with error -5.
                   at: (drivers/vulkan/rendering_device_vulkan.cpp:1605)
                ERROR: vmaMapMemory failed with error -5.
                   at: (drivers/vulkan/rendering_device_vulkan.cpp:1605)
                ERROR: vmaMapMemory failed with error -5.
                   at: (drivers/vulkan/rendering_device_vulkan.cpp:1605)
                ERROR: vmaMapMemory failed with error -5.
                   at: (drivers/vulkan/rendering_device_vulkan.cpp:1605)

is this normal after every start:
                Godot Engine v4.0.dev.20211210.official.21b81998b - https://godotengine.org
                vkEnumerateInstanceVersion not available, assuming Vulkan 1.0.
                Vulkan API 1.0.0 - Using Vulkan Device #0: AMD - AMD Radeon(TM) R4 Graphics

### Steps to reproduce

just create a new project on my pc...

**Comment by Calinou** [comment]
> is this normal after every start:

`vkEnumerateInstanceVersion` should be available on modern GPUs, but it may not be implemented on old GPUs that have partial Vulkan support.

Try updating your AMD graphics drivers to the latest version *provided by AMD* (not your laptop OEM).

**Comment by hackerManik3** [comment]
vkEnumerateInstanceVersion error was fixed after the updating my drivers but still cannot create new projects due to this error
            
ERROR: vmaMapMemory failed with error -5.
               at: (drivers/vulkan/rendering_device_vulkan.cpp:1605)

**Comment by hackerManik3** [comment]
> > is this normal after every start:
> 
> `vkEnumerateInstanceVersion` should be available on modern GPUs, but it may not be implemented on old GPUs that have partial Vulkan support.
> 
> Try updating your AMD graphics drivers to the latest version _provided by AMD_ (not your laptop OEM).

vkEnumerateInstanceVersion error was fixed after the updating my drivers but still cannot create new projects due to this error
and CRASH :-(

ERROR: vmaMapMemory failed with error -5.
at: (drivers/vulkan/rendering_device_vulkan.cpp:1605)

in drivers/vulkan/rendering_device_vulkan.cpp line 1605, i found this code:

           //map staging buffer (It's CPU and coherent)

	void *data_ptr = nullptr;
	{
		VkResult vkerr = vmaMapMemory(allocator, staging_buffer_blocks[staging_buffer_current].allocation, &data_ptr);
		ERR_FAIL_COND_V_MSG(vkerr, ERR_CANT_CREATE, "vmaMapMemory failed with error " + itos(vkerr) + ".");
	}

**Comment by MuffinInACup** [comment]
Godot4.0 beta3 - get the same error on launch; white screen and crash later

Verbose output:

```Godot Engine v4.0.beta3.official.01ae26d31 - https://godotengine.org
Text-to-Speech: SAPI initialized.
vkEnumerateInstanceVersion not available, assuming Vulkan 1.0.
Vulkan devices:
  #0: NVIDIA GeForce GTX 1060 - Supported, Discrete
  #1: Intel Intel(R) HD Graphics 630 - Supported, Integrated
Vulkan API 1.0.0 - Using Vulkan Device #0: NVIDIA - GeForce GTX 1060
- Vulkan Variable Rate Shading not supported
- Vulkan multiview supported:
  max view count: 32
  max instances: 134217727
- Vulkan subgroup:
  size: 0
  stages:
  supported ops:
Using present mode: VK_PRESENT_MODE_FIFO_KHR
Using "winink" pen tablet driver...
Creating VMA small objects pool for memory type index 7
Shader 'CanvasSdfShaderRD' SHA256: c232900d90b0aa53670c1ae4f73e1f6f24c4e84c4d8fa033643206613fb77225
Shader 'SkeletonShaderRD' SHA256: e085745a08805fc3ad41f488555b708666f8f8494c855a1dbf8ddf2e113d6742
Shader 'ParticlesShaderRD' SHA256: 643cff3b8d61f5823e51f27f16efcdf2f7a3d442b523580fec0d9eed35de6a3e
ERROR: vkCreateComputePipelines failed with error -1000012000.```
